File: FindKthBitinNthBinaryString.cpp
class Solution {
public:
    string BitString(int n, string s)
    {
        if(n == 0)
            return s;
        int a = s.size();
        string st = s;
        for(int i = 0 ; i < a ; i++)
        {
            if(st[i] == '0')
                st[i] = '1';
            else if(st[i] == '1')
                st[i] = '0';
        }
        for (int i = 0; i < a / 2; i++)
            swap(st[i], st[a - i - 1]);
        
        s = s + "1" + st;
        return BitString(--n, s);
    }
    
    char findKthBit(int n, int k) {
        string s = "0" ;
        s = BitString(n,s);
        return s[k-1];
    }
};

File: FindKthBitInNthBinaryString.py
class Solution:
    def operation(self,arr):
        newArr = arr.copy()
        newArr.append(1)
        for i in range(len(arr)):
            arr[i] = (1-arr[i])
        
        arr.reverse()
        newArr.extend(arr)
        return newArr

    def find(self,k,arr):
        if k <= len(arr):
            return arr[k-1]

        n = len(arr)
        newArr = self.operation(arr)
        return self.find(k,newArr)

    def findKthBit(self, n: int, k: int) -> str:
        return str(self.find(k,[0]))
